digdag-standards/src/main/java/io/digdag/standards/operator/aws/EmrOperatorFactory.java

```diff
@@ -0,0 +1,1739 @@
+        this.objectMapper = objectMapper;
+        this.cf = cf;
+        this.environment = environment;
+    }
+
+    @Override
+    public Operator newOperator(Path projectPath, TaskRequest request)
+    {
+        return new EmrOperator(projectPath, request);
+    }
+
+    private class EmrOperator
+            extends BaseOperator
+    {
+        private final TaskState state;
+        private final Config params;
+        private final String defaultActionOnFailure;
+
+        EmrOperator(Path projectPath, TaskRequest request)
+        {
+            super(projectPath, request);
+            this.state = TaskState.of(request);
+            this.params = request.getConfig()
+                    .mergeDefault(request.getConfig().getNestedOrGetEmpty("aws").getNestedOrGetEmpty("emr"))
+                    .mergeDefault(request.getConfig().getNestedOrGetEmpty("aws"));
+            this.defaultActionOnFailure = params.get("action_on_failure", String.class, "CANCEL_AND_WAIT");
+        }
+
+        @Override
+        public List<String> secretSelectors()
+        {
+            // TODO: extract secret references from step configurations
+            return ImmutableList.of("*");
+        }
+
+        @Override
+        public TaskResult run(TaskExecutionContext ctx)
+        {
+            String tag = state.constant("tag", String.class, EmrOperatorFactory::randomTag);
+
+            AWSCredentials credentials = credentials(tag, ctx);
+
+            SecretProvider awsSecrets = ctx.secrets().getSecrets("aws");
+
+            Optional<String> emrEndpoint = awsSecrets.getSecretOptional("emr.endpoint");
+            Optional<String> s3Endpoint = awsSecrets.getSecretOptional("s3.endpoint");
+            Optional<String> kmsEndpoint = awsSecrets.getSecretOptional("kms.endpoint");
+
+            ClientConfiguration emrClientConfiguration = new ClientConfiguration();
+            ClientConfiguration s3ClientConfiguration = new ClientConfiguration();
+            ClientConfiguration kmsClientConfiguration = new ClientConfiguration();
+
+            Aws.configureProxy(emrClientConfiguration, emrEndpoint, environment);
+            Aws.configureProxy(s3ClientConfiguration, s3Endpoint, environment);
+            Aws.configureProxy(kmsClientConfiguration, kmsEndpoint, environment);
+
+            AmazonElasticMapReduce emr = new AmazonElasticMapReduceClient(credentials, emrClientConfiguration);
+            AmazonS3Client s3 = new AmazonS3Client(credentials, s3ClientConfiguration);
+            AWSKMSClient kms = new AWSKMSClient(credentials, kmsClientConfiguration);
+
+            // Set up file stager
+            Optional<AmazonS3URI> staging = params.getOptional("staging", String.class).transform(s -> {
+                AmazonS3URI uri = new AmazonS3URI(s);
+                if (uri.getKey() != null && !uri.getKey().endsWith("/")) {
+                    throw new ConfigException("Invalid staging uri: '" + s + "'");
```

Comment:
Don't we need to add something like "The URI should have a tailing slash" here or on the document?